yara-x-dump/src/tests/testdata/macho_x86_file.out

```diff
@@ -0,0 +1,13 @@
+field1: 0x7b
+field2: "test"
+# Nested segments
+segments:
+  - nested1: 456
+    nested2: 0x315
+    timestamp: 123456789 # 1973-11-29 21:33:09 UTC
+# Nested optional
+optional:
+  - onested1: 123
```

Comment:
There's something wrong with this output. The `optional` field is declared in the proto as:

```
optional OptionalNested optional = 4;
```

So, it's not a repeated field, is just a nested structure. However, in the YAML output it's represented as an array. The correct output should be:

```yaml
segments:
  - nested1: 456
    nested2: 0x315
    timestamp: 123456789 # 1973-11-29 21:33:09 UTC
optional:
  onested1: 123
  onested2: 0x1c8
  map_string_string:
        foo: "bar"
```

Notice that there's no `-` in front of `onested1: 123`. This `-` indicates that that's an item in a list, and therefore `optional` is a list.

Reply:
Also I miss some tests with multi-line string values. For example, if the value for key "foo" in `map_string_string` is "foo\nbar" instead of just "bar", what happens? That's where things starts getting hairy. Make sure the result is not:

```yaml
optional:
  onested1: 123
  onested2: 0x1c8
  map_string_string:
        foo: "foo
bar"
```

Reply:
>Notice that there's no - in front of onested1: 123. This - indicates that that's an item in a list, and therefore optional is a list.

Thank you for noticing. I did not notice this and the validity YAML check passed. I will fix this

>Also I miss some tests with multi-line string values. For example, if the value for key "foo" in map_string_string is "foo\nbar" instead of just "bar", what happens? That's where thing starts getting hairy. Make sure the result is not:

I will add those tests, this is probably a bug from latest change where I wanted to make string coloured and reverted from using string escaping from protobuf library. I will take a look at it.